src/vcpkg/base/system.cpp

```diff
@@ -325,7 +325,22 @@ namespace vcpkg
         return ProgramW6432;
     }
 
-    int get_num_logical_cores() { return std::thread::hardware_concurrency(); }
+    int get_concurrency()
+    {
+        static int concurrency = [] {
+            auto user_defined_concurrency = get_environment_variable("VCPKG_MAX_CONCURRENCY");
+            if (user_defined_concurrency)
+            {
+                return std::stoi(user_defined_concurrency.value_or_exit(VCPKG_LINE_INFO));
+            }
+            else
+            {
+                return static_cast<int>(std::thread::hardware_concurrency()) + 1;
```

Comment:
maybe `return max(1, static_cast<int>(std::thread::hardware_concurrency());` why add another thread?

Reply:
Fwiw this is not a behavior change; this is the behavior we had before.

I believe this comes from the apocryphal idea that the compiler threads will sometimes be cpu-bound and sometimes be io-bound and thus you want to oversubscribe to ensure you saturate the cpu (so 1 thread at a time can be waiting for i/o).